core/codegen/src/ident.rs

```diff
@@ -0,0 +1,128 @@
+//! A variant of Ident with more ergonomic handling of raw identifiers.
+
+use std::cmp::Ordering;
+
+use devise::syn;
+use self::syn::ext::IdentExt as _;
+use crate::proc_macro2::{Ident as Ident2, Span};
+use quote::ToTokens;
+
+#[derive(Clone, Debug)]
```

Comment:
We should have a doc-string explaining why this exists and what it intends to do.